src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/cas2/ApplicationNoteService.kt

```diff
@@ -46,13 +53,32 @@ class ApplicationNoteService(
 
     val savedNote = saveNote(application, note.note, user)
 
+    if (isExternalUser) {
+      sendEmailToReferrer(application, savedNote)
+    }
+
     return AuthorisableActionResult.Success(
       ValidatableActionResult.Success(
         savedNote,
       ),
     )
   }
 
+  private fun sendEmailToReferrer(application: Cas2ApplicationEntity, savedNote: Cas2ApplicationNoteEntity) {
+    emailNotificationService.sendEmail(
+      // TODO should we check if email exists? Throw error if it doesn't
```

Comment:
Yep that's a good point that it should be handled further up stream...
It might be a bit tricky to do it when they first sign in because of the way we integrate with HMPPS Auth - but CAS1 have a solution where the user can update their details via the application form which we could use. 

It's a bit complicated as I think we would  have to emit a Domain Event as well, so that Nomis could be updated as well, but would be possible. I've made a ticket in the backlog here https://dsdmoj.atlassian.net/browse/CAS2-323

Reply:
I've said on the card that it might be worth asking the Nomis team if any POMs in their service do have missing emails... I wonder if the nullable nature of it is just a technicality, and in practice they have to have an email to sign up to Nomis.